File: momose/src/engine/AiManager.java
package engine;

import java.util.Iterator;
import java.util.Vector;

import models.Model;

public class AiManager 
 {
  private Vector thinkers; 	
  private Vector thinkersModels; 
	
  public AiManager()
   {
	thinkers=new Vector();
	thinkersModels=new Vector();  
   }//Fine costruttore
  
  public void addThinkers(Vector newThinkers)
   {thinkers.addAll(newThinkers); }
  
  public void addThinkerModel(Model newThinkerModel)
   {thinkersModels.add(newThinkerModel); }
  
  public int getNumThinkers()
   {return thinkers.size(); }
 
  public int getNumModThinkers()
   {return thinkersModels.size(); }
  
  
  public void animateAI(SimTime time)
   {
	//Faccio pensare i modelli registrati
	Iterator j=thinkersModels.iterator();
	while(j.hasNext())
	 {
	  ThinkerObject thinker=(ThinkerObject)j.next();
	  //Faccio pensare il thinker
	  thinker.think(time);
	 } 	  
	  
	//Faccio pensare i nodi registrati  
	Iterator i=thinkers.iterator();
	while(i.hasNext())
	 {
	  ThinkerObject thinker=(ThinkerObject)i.next();
	  //Faccio pensare il thinker
	  thinker.think(time);
	 } 	
   }//Fine animateAI
  
 }//Fine AIMAnager

File: momosecpp/src/AiManager.cpp
#include "AiManager.h"

AiManager::AiManager()
 {
  thinkers=new vector<Node*>();
  thinkersModels=new vector<Model*>(); 	
 }//Fine costruttore
 
void AiManager::addThinkers(vector<Node*> *newThinkers)
 {
  vector<Node*>::iterator i;
   for(i=newThinkers->begin();i!=newThinkers->end();i++)
    {thinkers->push_back(*i);}
 }//Fine addThinkers
 
void AiManager::animateAI(SimTime *simTime)
 {
  //Faccio pensare i modelli	
  ThinkerObject *modThinker;	
  vector<Model*>::iterator i;
   for(i=thinkersModels->begin();i!=thinkersModels->end();i++)
	{
	 modThinker=*i;
	 //Faccio pensare il thinker
	 modThinker->think(simTime);
	} 	  
	  
	//Faccio pensare i nodi registrati  
	ThinkerObject *nodeThinker;	
    vector<Node*>::iterator j;
    for(j=thinkers->begin();j!=thinkers->end();j++)
	 {
	  ThinkerObject *nodeThinker=*j;
	  //Faccio pensare il thinker
	  nodeThinker->think(simTime);
	 } 	  
 }//Fine animateAI

AiManager::~AiManager(){}
